BUILDBOT.md

```diff
@@ -0,0 +1,134 @@
+> [!IMPORTANT]  
+> To connect to WebKit infrastructure credentials are required.
+> Contact the infrastructure team on [Slack](https://webkit.slack.com) for
+> access.
+
+## Getting the image
+The `webkitdev/buildbot-worker` image is built and pushed to
+[Docker Hub](https://hub.docker.com/r/webkitdev/buildbot-worker). The following
+tags are supported. For the latest information on Windows container version
+compatibility see the
+[documentation](https://learn.microsoft.com/en-us/virtualization/windowscontainers/deploy-containers/version-compatibility)
+
+| Tag Name  | Automated | Description |
+|---|:---:|---|
+| 2022 | :white_check_mark: | A Windows 2022 server container |
+| windows-2022 | :x: | A Windows container, used for Layout Tests |
+| 1809 | :x: | A Windows 2019 server container |
+
+Visit [Docker Hub](https://hub.docker.com/r/webkitdev/buildbot-worker/tags) to
+see when the last build happened for the image's tags.
+
+## Configuring the Buildbot worker
+The `webkitdev/buildbot-worker` image uses environment variables for
+configuration. When invoking `docker run` environment variables are specified on
+the command line individually or through a file. See the
+[documentation](https://docs.docker.com/engine/reference/commandline/container_run/#env)
+for more details.
+
+| Environment Variable  | Required | Description |
+|---|:---:|---|
+| BUILD_HOST_NAME | :white_check_mark: | Host of the buildbot instance, either `build.webkit.org` or `ews-build.webkit.org` |
+| BUILD_HOST_PORT | :x: | Port of the buildbot instance, defaults to 17000 |
+| BUILD_WORKER_NAME | :white_check_mark: | Name for the worker, provided by infrastructure team  |
+| BUILD_WORKER_PASSWORD | :white_check_mark: | Password for the worker, provided by infrastructure team |
+| BUILD_WORKER_KEEPALIVE | :x: | Time in seconds that messages should be sent by the worker to the server, defaults to 240 |
+| ADMIN_NAME | :x:| Contact name for the admin of the worker | 
+| ADMIN_EMAIL | :x: | Contact e-mail address for the admin of the worker |
+| HOST_DESCRIPTION |:x: | A description of the host running the worker |
+| COMPILER | :x: | Set this to `Clang` to use `clang-cl`, defaults to `cl` |
+
+The preferred method of setup uses an environment file. An example file to fill
+out with all the options present looks like this. Save the text file locally
+with the name corresponding to worker name; for this example `build.env` is
+used.
+
+```txt
+ADMIN_EMAIL=<email>
+ADMIN_NAME=<admin>
+BUILD_HOST_NAME=<host>
+BUILD_WORKER_NAME=<name>
+BUILD_WORKER_PASSWORD=<password>
+HOST_DESCRIPTION=<description>
+COMPILER=Clang
+```
+
+## Running the Buildbot worker
+On a Windows Server 2022 machine run the following in a powershell session. On
+Windows Server 2022 the Docker container runs in process isolation mode which
+gives it access to all the resources of the host.
+
+```powershell
+docker run --name build --env-file build.env --detach --restart always webkitdev/buildbot-worker:2022
```

Comment:
The tag connection is a bit tenuous here, with the docs like this I'd expect people to end up using this even on a layout tester and then have trouble figuring out why it's failing.